### DISCUSSION TITLE
Use multistep Form for update

### DISCUSSION THREAD

**Discussion by spiii** [Budibase/budibase#9614] [Help]
Can someone provide a solution to achieve an update form as multistep form.

It seems not to be possible for me.
I have multiple tables with some relationship to each other.
It is very cool to use multistep Form for creation of them.
The problem is the update. I can update all the tables separately, but not at once with the multistep form.

I have no idea how to bind multiple data sources (because multiple separate tables) to the multistep Form.

Is there someone with any idea for that?

Thanks in advance

**Comment by melohagan** [comment]
Hey @spiii 

What you could do is use [App state](https://docs.budibase.com/docs/app-state) to save the **_id** of the row being display in your given form step when you click next. 

Then you can use multiple **Save Row** actions in your final 'Submit' step to update each row. Or you could simply do a **Save Row** as each next step button is clicked.

The downside is that you need to use a custom form schema, and `Add columns` manually when doing the save row. That's because currently there is no way to support multiple schemas in a form step setup, hence the single custom schema. 

There's also this issue which may be related: https://github.com/Budibase/budibase/issues/7229
